frontend/pkg/frontend/middleware_logging.go

```diff
@@ -71,13 +73,45 @@ func MiddlewareLogging(w http.ResponseWriter, r *http.Request, next http.Handler
 	logger = logger.With(
 		"request_method", r.Method,
 		"request_path", r.URL.Path,
+	)
+
+	// make a best attempt at parsing the resourceID. This will often fail because we have non-resource requests.
+	// we do this so that we can add subscription, resourceGroup, and hcpCluster to the logger context for future searching
+	// if possible.
+	// It's important to do before the second panic handler so that panics can be correlated easily.
+	// TODO are the value we find case sensitive or case insensitive.  They used to be case sensitive, so I have left that
+	if resourceID, err := azcorearm.ParseResourceID(r.URL.Path); err == nil {
+		logger = logger.With(
+			"subscription_id", resourceID.SubscriptionID,
+			"resource_group", resourceID.ResourceGroupName,
+		)
```

Comment:
Should we have a `"resource_id"` attribute as well?  [We do in the backend](https://github.com/Azure/ARO-HCP/blob/1b060917ac40f43c98b345d6d06c24f671e77311/backend/operations_scanner.go#L440), so that would be a common field name to search.  (Note, backend also has `"subscription_id"` but not `"resource_group"`.)

Reply:
> Should we have a "resource_id" attribute as well? [We do in the backend](https://github.com/Azure/ARO-HCP/blob/1b060917ac40f43c98b345d6d06c24f671e77311/backend/operations_scanner.go#L440), so that would be a common field name to search. (Note, backend also has "subscription_id" but not "resource_group".)

oh, I see. The idea that we could do a best effort here to help earlier in the stack?  I'd say it's a good followup.